src/App.tsx

```diff
@@ -1,14 +1,45 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [statusSelect, setStatusSelect] = useState('all');
+
+  let visibleTodos = [...todos].filter(todo => (
+    todo.title.toLowerCase().includes(query.toLowerCase().trim())));
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      });
+  }, []);
+
+  if (statusSelect !== Status.All) {
+    visibleTodos = visibleTodos.filter((todo) => {
+      switch (statusSelect) {
+        case Status.Active:
+          return !todo.completed;
+        case Status.Completed:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+  }
```

Comment:
instead of modifying variable `useMemo` can be used
```suggestion
  const visibleTodos = useMemo(() => [...todos]
    .filter(todo => (
      todo.title.toLowerCase().includes(query.toLowerCase().trim())
    )).filter((todo) => {
      switch (statusSelect) {
        case Status.Active:
          return !todo.completed;
        case Status.Completed:
          return todo.completed;
        default:
          return true;
      }
    }), [todos, statusSelect, query]);
```